### PR TITLE
[C2B2/TimeLock Serialization] Phase 2b: Days of Ruin (Client-Side)

### PR DESCRIPTION
**Pull request by jeremyk-91**
- [x] Needs dependency bump after merge of #6160 

## General
**Before this PR**:
Clients call refresh and unlock with a large lock token

**After this PR**:

==COMMIT_MSG==
Clients call refresh and unlock with smaller lock tokens.
==COMMIT_MSG==

**Priority**: P1

**Concerns / possible downsides (what feedback would you like?)**:
I opted to make a relatively smaller change than I did in the Hack Week version of this PR. In particular, I kept the old `ConjureLockToken` around in e.g. `LeasedLockToken`, just rewriting that in `LockLeaseService` to be a V2 token whenever we actually interact with timelock. Is this correct?

**Is documentation needed?**: No

## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**: No API breaks.

**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**: No

**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**: Yes

**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**: Requires TimeLock after the first PR: #6160 

**Does this PR need a schema migration?** No

## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**: Not much?

**What was existing testing like? What have you done to improve it?**: Integration tests will now be using the V2 endpoints.

**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**: No

**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**: No

## Execution
**How would I tell this PR works in production? (Metrics, logs, etc.)**: Request rates on Timelock, and to the ConjureBlockingTimelockService

**Has the safety of all log arguments been decided correctly?**: Neither `SafeArg` nor `UnsafeArg` appears in our search

**Will this change significantly affect our spending on metrics or logs?**: No

**How would I tell that this PR does not work in production? (monitors, etc.)**: Standard errors over time, also if metrics don't change the way we want

**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**: Yep, just rollback

**If the above plan is more complex than “recall and rollback”, please tag the support PoC here (if it is the end of the week, tag both the current and next PoC)**:

## Scale
**Would this PR be expected to pose a risk at scale? Think of the shopping product at our largest stack.**: Not really, this is expected to be a strict improvement

**Would this PR be expected to perform a large number of database calls, and/or expensive database calls (e.g., row range scans, concurrent CAS)?**: No

**Would this PR ever, with time and scale, become the wrong thing to do - and if so, how would we know that we need to do something differently?**: I don't think so. Unless we need to package some information with the lock token

## Development Process
**Where should we start reviewing?**: It's small

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**: It's not

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@sverma30
@raiju

### CODE REVIEW

File: lock-api/src/main/java/com/palantir/lock/client/LockLeaseService.java [hunk 1]
```diff
@@ -164,12 +166,14 @@ private Set<LeasedLockToken> refreshTokens(Set<LeasedLockToken> leasedTokens) {
...
-                delegate.refreshLocks(ConjureRefreshLocksRequest.of(serverTokens(leasedTokens)));
+        ConjureRefreshLocksResponseV2 refreshLockResponse =
+                delegate.refreshLocksV2(ConjureRefreshLocksRequestV2.of(serverTokens(leasedTokens)));
         Lease lease = refreshLockResponse.getLease();
 
         Set<LeasedLockToken> refreshedTokens = leasedTokens.stream()
-                .filter(t -> refreshLockResponse.getRefreshedTokens().contains(t.serverToken()))
+                .filter(t -> refreshLockResponse
+                        .getRefreshedTokens()
+                        .contains(LockLeaseService.convertV1Token(t.serverToken())))
```
Comment (gmaretic) [inline lock-api/src/main/java/com/palantir/lock/client/LockLeaseService.java:176]: Is there a reason why converting to v1 is more beneficial than just having LeasedLockToken use v2?

File: lock-api/src/main/java/com/palantir/lock/client/NamespacedConjureTimelockService.java [hunk 2]
```diff
@@ -34,8 +38,12 @@
 public interface NamespacedConjureTimelockService {
     ConjureUnlockResponse unlock(ConjureUnlockRequest request);
```
Comment (gmaretic) [inline lock-api/src/main/java/com/palantir/lock/client/NamespacedConjureTimelockService.java:39]: Is there a reason to keep this API? I get that breaking APIs is a "bad thing"™️, but we actively don't want usages after this change

File: lock-api/src/main/java/com/palantir/lock/client/LockLeaseService.java (hunk 1 above)
Comment (gmaretic) [reply lock-api/src/main/java/com/palantir/lock/client/LockLeaseService.java:176]: Ah I see you actually mention this. This seems correct, but janky -- I'd prefer the full change, but I am also ok with leaving that for a follow-up PR

**Review by gmaretic — APPROVED** [review]
Wiring looks correct, and we are indeed always using the new endpoints now. I do think we should change LeasedLockToken, and I would also break the API to force users to use the new endpoints -- but neither of these is  a strictly better approach, so I am happy mentioning it and letting you make the decision

File: changelog/@unreleased/pr-6149.v2.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+type: improvement
+improvement:
+  description: |-
+    TimeLock exposes three new endpoints:
```
[the remark this answers is not in the document]
Comment (jeremyk-91) [reply changelog/@unreleased/pr-6149.v2.yml:4]: ugh. Yeah, that's right - will fix

File: lock-api/src/main/java/com/palantir/lock/client/NamespacedConjureTimelockService.java (hunk 2 above)
Comment (jeremyk-91) [reply lock-api/src/main/java/com/palantir/lock/client/NamespacedConjureTimelockService.java:39]: I think the intention was to make the smallest possible change given tight timelines, though happy to do this in a follow up

File: lock-api/src/main/java/com/palantir/lock/client/LockLeaseService.java (hunk 1 above)
Comment (jeremyk-91) [reply lock-api/src/main/java/com/palantir/lock/client/LockLeaseService.java:176]: Yeah, let's do that in a follow up, given this is time sensitive

**Pull request merged**